docs/cloud-provider-specifics.md

```diff
@@ -181,75 +185,108 @@ Now that you have your Google Cloud credentials stored in a Secret, you need to
+4. Set the name of the Resource Group that contains your Kubernetes cluster.
 
+    ```bash
+    # Change "Kubernetes" as needed
+    AZURE_RESOURCE_GROUP=Kubernetes
     ```
-    az group list
-    ```
-     Get your cluster's group `name` from the response, and use it to set `$AZURE_RESOURCE_GROUP`. (Also note the `location`--this is later used in the Azure-specific portion of the Ark Config).
 
-5. Create a service principal with the "Contributor" role:
+    If you are unsure of the Resource Group name, run the following command to get a list that you can select from. Then set the `AZURE_RESOURCE_GROUP` environment variable to the appropriate value.
 
+    ```bash
+    az group list --query '[].{ ResourceGroup: name, Location:location }'
     ```
-    az ad sp create-for-rbac --role="Contributor" --name="heptio-ark"
-    ```
-    From the response, save `appId` into `$AZURE_CLIENT_ID` and `password` into `$AZURE_CLIENT_SECRET`.
 
-6. Login into the `heptio-ark` service principal account:
+    Get your cluster's group `name` from the response, and use it to set `$AZURE_RESOURCE_GROUP`. (Also note the `location`--this is later used in the Azure-specific portion of the Ark Config).
 
-    ```
-    az login --service-principal \
-        --username http://heptio-ark \
-        --password $AZURE_CLIENT_SECRET \
-        --tenant $AZURE_TENANT_ID
-    ```
+5. Create a service principal with `Contributor` role. This will have subscription-wide access, so protect this credential. You can specify a password or let the `az ad sp create-for-rbac` command create one for you.
 
-7. Specify a *globally-unique* storage account id and save it in `$AZURE_STORAGE_ACCOUNT_ID`. Then create the storage account, specifying the optional `--location` flag if you do not have defaults from `az configure`:
+    ```bash
+    # Create service principal and specify your own password
+    AZURE_CLIENT_SECRET=super_secret_and_high_entropy_password_replace_me_with_your_own
+    az ad sp create-for-rbac --name "heptio-ark" --role "Contributor" --password $AZURE_CLIENT_SECRET
 
+    # Or create service principal and let the cli generate a password for you. ensure we capture the password though.
+    AZURE_CLIENT_SECRET=`az ad sp create-for-rbac --name "heptio-ark" --role "Contributor" --query 'password' -o tsv`
+
+    # After creating the service principal, obtain the client id
+    AZURE_CLIENT_ID=`az ad sp list --display-name "heptio-ark" --query '[0].appId' -o tsv`
     ```
-    az storage account create \
-        --name $AZURE_STORAGE_ACCOUNT_ID \
-        --resource-group $AZURE_RESOURCE_GROUP \
-        --sku Standard_GRS
-    ```
-    You will encounter an error message if the storage account ID is not unique; change it accordingly.
 
-8. Get the keys for your storage account:
+6. Create the storage account and blob container for Ark to store the backups in.
```

Comment:
We don't have a section for creating a bucket for AWS/GCP. Would be nice to be consistent across the 3.

Reply:
I'll file issues